unicon_backend/evaluator/tasks/programming/steps.py

```diff
@@ -266,7 +274,7 @@ def _serialize_data(data: str | int | float | bool) -> str:
 
         program: Program = [*self.debug_stmts()]
         for output in self.outputs:
-            assert output.data is not None
+            assert output.id.startswith("CONTROL") or output.data is not None
```

Comment:
```suggestion
            assert output.type == "CONTROL" or output.data is not None
```

Reply:
this seems to already guaranteed by `check_non_empty_outputs`?

Reply:
47563bb